ocaml/runtime/gc_ctrl.c

```diff
@@ -281,15 +265,10 @@ static value gc_full_major_exn(int force_compaction)
   /* In general, it can require up to 3 GC cycles for a
      currently-unreachable object to be collected. */
   for (i = 0; i < 3; i++) {
-<<<<<<< HEAD
-    caml_empty_minor_heaps_once();
+    // XXX mshinwell for sdolan: please check conflict diff here.
+    // Maybe we should be taking the upstream version now?  (I think we made
+    // a fix to ensure that Gc.compact () did what people expected)
```

Comment:
I think we should take upstream 5.2's version.

Both sides have the same fix, but implemented slightly differently. Let's go with upstream's to minimise the diff.